src/LanguageServer/Protocol/Handler/DocumentChanges/DidChangeHandler.cs

```diff
@@ -67,7 +67,7 @@ private static SourceText GetUpdatedSourceText(TextDocumentContentChangeEvent[]
         {
             // The changes were in reverse document order, so we can merge them into a single operation on the source text.
             // Note that the WithChanges implementation works more efficiently with it's input in forward document order.
-            var newChanges = contentChanges.Reverse().SelectAsArray(change => ProtocolConversions.ContentChangeEventToTextChange(change, text));
+            var newChanges = Enumerable.Reverse(contentChanges).SelectAsArray(change => ProtocolConversions.ContentChangeEventToTextChange(change, text));
```

Comment:
Don't forget to add `TypeForwardedTo(Enumerable)` in the modern scenarios to prevent tearing issues causing `MethodMissingException`s.

Reply:
I don't think I can do that now since the `Enumerable.Reverse(this T[])` does not exist yet on modern .NET.

Also for extension methods we shouldn't need `TypeForwardedTo` [per this doc](
https://github.com/dotnet/roslyn/blob/5a39d0ad691ee88dcf163808d9ed7a46b168a1c9/docs/contributing/Target%20Framework%20Strategy.md#pattern-for-extension-methods), we can only add non-extension method once the one in stdlib is available.